controllers/datanetwork_controller.go

```diff
@@ -72,31 +78,45 @@ func dataNetworkUpdateRequired(instance *starlingxv1.DataNetwork, n *datanetwork
 		if vxlan.EndpointMode != nil && n.Mode != nil {
 			if *vxlan.EndpointMode != *n.Mode {
 				opts.Mode = vxlan.EndpointMode
+				delta.WriteString(fmt.Sprintf("\t+Mode: %s\n", *opts.Mode))
 				result = true
 			}
 		}
 
 		if vxlan.UDPPortNumber != nil && n.UDPPortNumber != nil {
 			if *vxlan.UDPPortNumber != *n.UDPPortNumber {
 				opts.PortNumber = vxlan.UDPPortNumber
+				delta.WriteString(fmt.Sprintf("\t+PortNumber: %d\n", *opts.PortNumber))
 				result = true
 			}
 		}
 
 		if vxlan.TTL != nil && n.TTL != nil {
 			if *vxlan.TTL != *n.TTL {
 				opts.TTL = vxlan.TTL
+				delta.WriteString(fmt.Sprintf("\t+TTL: %d\n", *opts.TTL))
 				result = true
 			}
 		}
 
 		if vxlan.MulticastGroup != nil && n.MulticastGroup != nil {
 			if *vxlan.MulticastGroup != *n.MulticastGroup {
 				opts.MulticastGroup = vxlan.MulticastGroup
+				delta.WriteString(fmt.Sprintf("\t+MulticastGroup: %s\n", *opts.MulticastGroup))
 				result = true
 			}
 		}
 	}
+	deltaString := delta.String()
+	if deltaString != "" {
+		deltaString = "\n" + strings.TrimSuffix(deltaString, "\n")
+		logDataNetwork.Info(fmt.Sprintf("delta configuration:%s\n", deltaString))
+	}
+	instance.Status.Delta = deltaString
```

Comment:
Personally, I prefer to update status in ReconcileUpdated, so that we don't need to pass Reconciler here and easier to understand when and where status is updated. But I'd like to hear form the other reviewer.

Reply:
Hi Tak. Yes, I have also considered that option. The only reason I haven't implemented it that way is because in the ReconcileUpdated function, we have two return statements. So, perhaps we need to include the update status in both places.
I am open to modifying the code depending on @yjian118 's opinion.

Reply:
I'm okay with the current implementation as updating in this point explains the current config vs specs. And I believe it will get updated after the reconciliation in the next round of update_required check